cmake/ImportBSON.cmake

```
@@ -0,0 +1,170 @@
+     the MONGOCRYPT_MONGOC_DIR.
+   - mongo::mongocrypt will be built with a dynamic link requirement on a libbson dynamic
+     library, which must be resolved at runtime by consumers. The translation units from the
+     MONGOCRYPT_MONGOC_DIR *will not* be included in the mongo::mongocrypt library.
+   - The installed libbson_for_shared will dynamically link to a libbson on the user's system by
+     using a find_library() call.
+
+   In both of the above cases, libbson_for_static will require that the final consumer
+   provide their own definitions of the libbson symbols, regardless of the value
+   of USE_SHARED_LIBBSON.
+]]
+
+include(CheckCSourceCompiles)
+include(CMakePushCheckState)
+
+cmake_push_check_state ()
+   # Even though we aren't going to use the system's libbson, try to detect whether it has
+   # extra-alignment enabled. We want to match that setting as our default, for convenience
+   # purposes only.
+   find_path(SYSTEM_BSON_INCLUDE_DIR bson/bson.h PATH_SUFFIXES libbson-1.0)
+   if (SYSTEM_BSON_INCLUDE_DIR AND NOT DEFINED ENABLE_EXTRA_ALIGNMENT)
+      set(CMAKE_REQUIRED_INCLUDES "${SYSTEM_BSON_INCLUDE_DIR}")
+      set (_extra_alignment_default OFF)
+      check_c_source_compiles([[
+         #include <bson/bson.h>
+
+         int main() { }
+      ]] HAVE_SYSTEM_LIBBSON)
+
+      if (HAVE_SYSTEM_LIBBSON)
+         # We have a libbson, check for extra alignment
+         check_c_source_compiles([[
+            #include <bson/bson.h>
+
+            #ifndef BSON_EXTRA_ALIGN
+            #error "Not extra-aligned"
+            #endif
+
+            int main() {}
+         ]] SYSTEM_LIBBSON_IS_EXTRA_ALIGNED)
+         if (SYSTEM_LIBBSON_IS_EXTRA_ALIGNED)
+            # Extra aligned! We'll use extra alignment by default.
+            set (_extra_alignment_default ON)
+         endif ()
+      endif ()
+   endif ()
+cmake_pop_check_state()
+
+set (init OFF)
+if (DEFINED ENABLED_SHARED_BSON)
+   message (STATUS "ENABLE_SHARED_BSON is now named USE_SHARED_LIBBSON")
+   set (init "${ENABLE_SHARED_BSON}")
+endif ()
+option (USE_SHARED_LIBBSON "Dynamically link libbson for the libmongocrypt dynamic library (default is static)" ${init})
+
+if (NOT DEFINED MONGOCRYPT_MONGOC_DIR)
+   # The user did not provide a MONGOCRYPT_MONGOC_DIR, so we'll get one
+   include (FetchContent OPTIONAL)
+   if (NOT COMMAND FetchContent_Declare)
+      # We need FetchContent in order to download the project.
+      message (FATAL_ERROR
+            "No MONGOCRYPT_MONGOC_DIR setting was defined, and the FetchContent.cmake "
+            "module is not available. Upgrade your CMake version, or provide a "
+            "MONGOCRYPT_MONGOC_DIR path to a mongo-c-driver directory (This is required "
+            "for libmongocrypt to find a libbson to use and link against).")
+   endif ()
+   include (FetchMongoC)
+   # The FetchMongoC module defines a MONGOCRYPT_MONGOC_DIR for use to use
```

It just hit me what you probably meant to say here.

```suggestion
   # The FetchMongoC module defines a MONGOCRYPT_MONGOC_DIR for us to use
```